instaclustr/resource_cluster.go

```diff
@@ -725,6 +709,64 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error{
 	return nil
 }
 
+func getBundlesFromCluster(cluster *Cluster) ([]map[string]interface{}, error) {
+	baseBundle := make(map[string]interface{}, 0)
+	baseBundle["bundle"] = cluster.BundleType
+
+	baseBundleOptions := make(map[string]interface{}, 0)
+	err := mapstructure.Decode(cluster.BundleOption, &baseBundleOptions)
+	if err != nil {
+		return nil, fmt.Errorf("[Error] Error decoding bundles option: %s", err)
+	}
+
+	convertedBundleOptions := convertInterface(baseBundleOptions)
+
+	baseBundle["options"] = convertedBundleOptions
+	baseBundle["version"] = cluster.BundleVersion
+
+	bundles := make([]map[string]interface{}, 0)
+	bundles = append(bundles, baseBundle)
+	if cluster.AddonBundles != nil {
+		bundles = append(bundles, cluster.AddonBundles)
+	}
+
+	return bundles, nil
+}
+
+func convertInterface(data map[string]interface{}) map[string]interface{} {
+	for k, v := range data {
+		// Terraform expects strings for everything
+		// This block iterates through the bundle options map and checks for an interface of a pointer
+		// For each interface{*type} value, it changes: interface{*type} -> *type -> type -> interface{type} -> String
+		// For a non-pointer, it directly formats to a string
+		valueOfV := reflect.ValueOf(v)
+		if valueOfV.Kind() == reflect.Ptr {
+			data[k] = fmt.Sprintf("%v", reflect.Indirect(valueOfV.Elem()).Interface())
+		} else {
+			data[k] = fmt.Sprintf("%v", valueOfV)
+		}
+	}
+
+	return data
+}
+
+func processProvider(d *schema.ResourceData, data map[string]interface{}) (newData map[string]interface{}) {
+	// This is used to ignore values that are not set in the resource
+	// Otherwise terraform will store the default and generated values (e.g. Provider Account Name)
+	// into the state and result a diff between the state and resource (plan)
+	newData = make(map[string]interface{})
+	for k, v := range data {
+		resource := d.Get(fmt.Sprintf("cluster_provider.%s", k))
+		// Store the required field "name"
+		// There doesn't seem to be a way to programmatically get the "required" schemas with ResourceData
```

Comment:
[requested] Perhaps we could store a list of requiredResources and iterate through them? Might make it more obvious in future how to add a required resource to this check

Reply:
As commented above this feels like a hacky solution and creating a list of requiredResources would add a duplication to our resource data that already exists so when we need to add more "required resource" check we should be attempting to do it differently (i.e. shouldn't encourage this solution)